### PR TITLE
Desktop 721

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

This implements the dumb component for the login with passphrase screen.

## 👻 stuff
When I actually try to use it I get this error:

screenshot_20160418-164659

## Preview

This lives in the stylesheet for now, will move to component sheet when we get one. 
screenshot_20160418-165247

### CODE REVIEW

File: shared/common-adapters/back-button.native.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ export default class BackButton extends Component {
   props: Props;
 
   onClick (event: SyntheticEvent) {
-    event.preventDefault()
-    event.stopPropagation()
+    event && event.preventDefault && event.preventDefault()
+    event && event.stopPropagation && event.stopPropagation()
```
Comment (MarcoPolo) [inline shared/common-adapters/back-button.native.js]: I got an error that `preventDefault` is not a fn

File: shared/login/register/passphrase/index.js [hunk 2]
```diff
@@ -5,14 +5,29 @@ import Render from './index.render'
...
     return <Render {...this.props}
       onForgotPassphrase={() => {
         this.props.onForgotPassphrase()
         this.props.onBack()
       }}
-      />
+      saveInKeychain={this.state.saveInKeychain}
+      showTyping={this.state.showTyping}
+      toggleShowTyping={showTyping => this.setState({showTyping})}
+      toggleSaveInKeychain={saveInKeychain => this.setState({saveInKeychain})}/>
```
Comment (MarcoPolo) [inline shared/login/register/passphrase/index.js]: the desktop doesn't use these yet, but I think it makes sense

File: shared/common-adapters/back-button.native.js (hunk 1 above)
Comment (chrisnojima) [inline shared/common-adapters/back-button.native.js]: hmmm, should be: https://facebook.github.io/react/docs/events.html

File: shared/login/register/passphrase/index.render.native.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+type State = {passphrase: string}
+
+class Render extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {passphrase: ''}
```
Comment (chrisnojima) [inline shared/login/register/passphrase/index.render.native.js]: could hoist this up...

File: shared/login/register/passphrase/index.render.native.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  onSubmit () {
+    if (this.state.usernameOrEmail) {
+      this.props.onSubmit(this.state.passphrase)
+    }
+  }
+
+  onChange (passphrase: string) {
+    console.log('p:', passphrase)
```
Comment (chrisnojima) [inline shared/login/register/passphrase/index.render.native.js]: 🔥

**Comment by chrisnojima** [comment]
👍 %c

File: shared/login/register/passphrase/index.render.native.js (hunk 4 above)
Comment (MarcoPolo) [inline shared/login/register/passphrase/index.render.native.js]: ‼️

**Pull request merged**